mass/core/query_handler.py

```diff
@@ -28,18 +33,87 @@ def __init__(
         self,
         *,
         aggregator_collection: AggregatorCollectionPort,
-        dao_collection: DaoCollectionPort
+        dao_collection: DaoCollectionPort,
     ):
         """Initialize the query handler with resource daos/aggregators"""
         self._aggregator_collection = aggregator_collection
         self._dao_collection = dao_collection
 
+    def _get_nested_fields(self, *, fields: List) -> Set:
+        """
+        Get nested fields from a given set of fields.
+
+        Args:
+            fields: A list of fields
+
+        Returns:
+            A set of nested fields
+
+        """
+        nested_fields = set()
+        for field in fields:
+            if "." in field:
+                top_level_field, nested_field = field.split(".", 1)
+                nested_fields.add((top_level_field, nested_field))
+        return nested_fields
+
+    def _pipeline_match_text_search(self, *, query: str) -> JsonObject:
+        """Build text search segment of aggregation pipeline"""
+        text_search = {"$text": {"$search": query}}
+        return {"$match": text_search}
+
+    def _pipeline_match_filters_stage(
+        self, *, filters: list[models.Filter]
+    ) -> JsonObject:
+        """Build segment of pipeline to apply search filters"""
+        segment: Dict = {}
+        for item in filters:
+            filter_key = "content." + str(item.key)
+            filter_value = item.value
+            try:
+                segment[filter_key]["$in"].append(filter_value)
+            except KeyError:
+                segment[filter_key] = {"$in": [filter_value]}
```

Comment:
You can use defaultdict from collections:

```suggestion
        segment: dict[list[str]] = defaultdict(list)
        for item in filters:
            filter_key = "content." + str(item.key)
            filter_value = item.value
            segment[filter_key]["$in"].append(filter_value)
```